File: 贪心算法/45.跳跃游戏II.go
func jump(nums []int) int {
	cur := 0
	nextCur := 0
	res := 0
	for i := 0; i < len(nums)-1; i++ {
		nextCur = max(nextCur, i+nums[i])
		if i == cur {
			cur = nextCur
			res++
		}
	}
	return res
}

File: 贪心算法/45.跳跃游戏II.java
class Solution {
    public int jump(int[] nums) {
        int cur=0,nextCur=0;
        int res=0;
        for(int i=0;i<nums.length-1;i++){
            nextCur=Math.max(nextCur,i+nums[i]);
            if(i==cur){
                cur=nextCur;
                res++;
            }
        }
        return res;
    }
}